### PR TITLE
Remove reasons from PodConditionType

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?

/kind cleanup
/sig api-machinery

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kerthcet** [comment]
/sig scheduling
cc @Huang-Wei

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -2543,12 +2543,6 @@ const (
 	PodReady PodConditionType = "Ready"
 	// PodInitialized means that all init containers in the pod have started successfully.
 	PodInitialized PodConditionType = "Initialized"
-	// PodReasonUnschedulable reason in PodScheduled PodCondition means that the scheduler
-	// can't schedule the pod right now, for example due to insufficient resources in the cluster.
-	PodReasonUnschedulable = "Unschedulable"
```
Comment (SataQiu) [inline pkg/apis/core/types.go:2548]: Given that these are exported types in the core API, don't we need to mark them as deprecated first and then safely remove them?

Comment (Huang-Wei) [reply pkg/apis/core/types.go:2548]: Generally we can suggest the users to use the literal from `k8s.io/api`, but not sure how to conventionally deprecate/remove this one (defined in k/k). Leaving the decision to sig-apimachinery.

Comment (Huang-Wei) [reply pkg/apis/core/types.go:2548]: A think a more general question asking advice for sig-apimachinery is: inside and outside k/k, shall we restrict accessing const literals from k/api only? If so, we can remove duplicated const literals from k/k#pkg/apis/.

Comment (kerthcet) [reply pkg/apis/core/types.go:2548]: weight in @liggitt for advices.

Comment (ahg-g) [reply pkg/apis/core/types.go:2548]: all components other than the api-server should depend on the staging repo only (i.e., the external type), I think those can be removed if not used by the api-server.

Comment (Huang-Wei) [reply pkg/apis/core/types.go:2548]: this variable is not used within k/k.

File: pkg/registry/core/pod/strategy.go [hunk 2]
```diff
@@ -686,7 +687,7 @@ func applyWaitingForSchedulingGatesCondition(pod *api.Pod) {
 	pod.Status.Conditions = append(pod.Status.Conditions, api.PodCondition{
 		Type:    api.PodScheduled,
 		Status:  api.ConditionFalse,
-		Reason:  api.PodReasonSchedulingGated,
+		Reason:  apiv1.PodReasonSchedulingGated,
```
Comment (Huang-Wei) [inline pkg/registry/core/pod/strategy.go:690]: Any side effect of changing it from internal or external types? (given this file belongs to apiserver cc @ahg-g )

File: pkg/apis/core/types.go (hunk 1 above)
Comment (kerthcet) [reply pkg/apis/core/types.go:2548]: cc @liggitt any concern here?

Comment (liggitt) [reply pkg/apis/core/types.go:2548]: the constants in the non-staging packages can be dropped if that makes sense, and we can switch things to reference the staging k8s.io/api exported constants

Comment (kerthcet) [reply pkg/apis/core/types.go:2548]: Thanks, can you approve this? Needs api approval.

**Pull request merged**